source/sync/data-access-patterns/flexible-sync-permissions-guide.txt

```diff
@@ -0,0 +1,611 @@
+  the document.
+- :guilabel:`Define Permissions`: use the template dropdown to
+  select the "Custom (start from scratch)". Paste the following into the rule
+  expression box: 
+
+  .. code-block:: json
+
+     {
+       "rules": {},
+       "defaultRoles": [
+         {
+           "name": "read-only",
+           "applyWhen": {},
+           "write": {
+             "$or": [
+               {
+                 "owner_id": "%%user.id"
+               },
+               {
+                 "collaborators": "%%user.id"
+               }
+             ]
+           }
+         }
+       ]
+     }
+  
+  Take a look at the "write" expression. ``$or`` takes an array of options. We
+  have two possible conditions where a user may write the document:
+  
+  - The ``owner_id`` field of the document equals the user's ID
+  - The ``collaborators`` array field of the document contains the user's ID.
+
+  Generally speaking, when a user is granted write permission, that user
+  automatically gets read permission. Therefore, we omitted the "read"
+  expression as it would be redundant in this case.
+
+
+Adding a Collaborator
+~~~~~~~~~~~~~~~~~~~~~
+
+A user can grant write access on their document to another user by adding that
+user's ID to the ``collaborators`` array field on their document. This can be
+done on the client side.
+
+Security
+~~~~~~~~
+
+We don't recommend using this model for highly sensitive data.
+
+This model uses the flexible sync permissions system to keep documents private
+between the document creator and the collaborators they add to the document.
+
+However, if a user has write access to a document, they may write to any field
+of the document. Consequently, this strategy **allows collaborators to add other
+collaborators**. It would also allow a collaborator to edit the ``owner_id``
+field. Finer control of permissions are not possible without field-level
+permissions, which are not currently available.
+
+User Search
+~~~~~~~~~~~
+
+How exactly to get the other user's ID depends on the details of your app. For
+example, when a user wants to add another user to a document, you might have a
+search box that accepts an email address. If the given email address corresponds
+to another user, the client can add that user's ID to the document's
+``collaborators`` array.
+
+Realm has no built-in way to search users. Generally, the flow for searching
+users is as follows:
+
+- Set up an authentication trigger to create a user document when a user
+  registers. The user document contains information you'll use to look up later,
+  such as the user's email address.
+- Create a function that queries the user data collection for a user.
+- Call the function from the client side when the user wants to find another user.
+
+To create an authentication trigger, follow these steps:
+
+.. include:: /includes/create-auth-trigger.rst
+
+- 
+   - 
+      .. code-block:: javascript
```

Comment:
there's 2 code blocks back to back here. 1 from the include and the one below. i believe this is a mistake.